crates/chain/tests/test_tx_graph.rs

```diff
@@ -822,3 +822,95 @@ fn test_additions_last_seen_append() {
         );
     }
 }
+
+/// Inserting anchors without the corresponding transaction
+/// and testing the recoverability of the transaction graph
+#[test]
+fn test_inserting_anchor_without_tx() {
+    let mut graph = TxGraph::<ConfirmationHeightAnchor>::default();
+    let mut graph_recovered = graph.clone();
+    let txs = (0..5).map(common::new_tx).collect::<Vec<_>>();
+    let chain: LocalChain = (0..5)
+        .map(|i| (i, h!(i.to_string())))
+        .collect::<BTreeMap<u32, BlockHash>>()
+        .into();
+    let anchors = (0..5)
+        .map(|i| ConfirmationHeightAnchor {
+            anchor_block: BlockId {
+                height: i,
+                hash: *chain.blocks().get(&i).unwrap(),
+            },
+            confirmation_height: 0,
+        })
+        .collect::<Vec<_>>();
+
+    let mut anchor_additions = Additions::default();
+    for i in 0..3 {
+        let additions = graph.insert_anchor(txs[i].txid(), anchors[i]);
+        anchor_additions.append(additions);
+    }
+
+    // Additions should only include the Anchor for the Tx-Block 0, 1 and 2
+    assert_eq!(
+        anchor_additions,
+        Additions {
+            tx: Default::default(),
+            txout: Default::default(),
+            anchors: BTreeSet::from([
+                (anchors[0], txs[0].txid()),
+                (anchors[1], txs[1].txid()),
+                (anchors[2], txs[2].txid()),
+            ]),
+            last_seen: Default::default(),
+        }
+    );
+
+    // graph should be recoverable from additions
+    graph_recovered.apply_additions(anchor_additions);
+    assert_eq!(graph, graph_recovered);
+
+    // Check whether Anchors are part of TxGraph
+    for i in 0..3 {
+        assert!(graph.all_anchors().contains(&(anchors[i], txs[i].txid())));
+    }
```

Comment:
This is another way of writing this, but I guess it's a matter of taste.

```suggestion
    assert_eq!(
        graph.all_anchors(),
        &anchors
            .iter()
            .cloned()
            .zip(txs.iter().map(Transaction::txid))
            .take(3)
            .collect()
    );
```